scripts/dotnet-cli-build/CompileTargets.cs

```diff
@@ -657,7 +655,6 @@ private static BuildTargetResult CompileCliSdk(BuildTargetContext c, DotNetCli d
                 {
                     // Yes, it is .exe even on Linux. This is the managed exe we're working with
                     File.Copy(Path.Combine(binaryToCorehostifyOutDir, $"{binaryToCorehostify}.exe"), Path.Combine(outputDir, $"{binaryToCorehostify}.dll"));
-                    File.Delete(Path.Combine(binaryToCorehostifyOutDir, $"{binaryToCorehostify}.exe"));
```

Comment:
Maybe we shouldn't be renaming `csc.exe` to `csc.dll` at all?  Maybe we should just leave it `csc.exe` - just like we are for `vbc.exe` now.

/cc @pakrym who added this code.